clang/lib/CodeGen/CGSYCLRuntime.h

```diff
@@ -0,0 +1,38 @@
+//===----- CGSYCLRuntime.h - Interface to SYCL Runtimes ---------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This provides custom clang code generation for SYCL.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CODEGEN_CGSYCLRUNTIME_H
+#define LLVM_CLANG_LIB_CODEGEN_CGSYCLRUNTIME_H
+
+#include "clang/AST/Decl.h"
+#include "llvm/IR/Value.h"
+
+namespace clang {
+namespace CodeGen {
+
+class CodeGenModule;
+
+class CGSYCLRuntime {
+protected:
+  CodeGenModule &CGM;
+
+public:
+  CGSYCLRuntime(CodeGenModule &CGM) : CGM(CGM) {}
+  virtual ~CGSYCLRuntime();
```

Comment:
whats wrong with the defaulted constructor?

Reply:
SYCL here follows the existing design of language-specific code generation "runtimes", such as CGOpenMPRuntime. They require CodeGenModule instance to work, hence no default constructor.

Reply:
Sorry, i meant destructor here.  Seems the destructor is completely unnecessary.

Reply:
@erichkeane , removed the destructor